eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Security/PolicyProvider/PoliciesConfigBuilder.php

```diff
@@ -14,28 +14,47 @@ class PoliciesConfigBuilder extends ContainerConfigBuilder
 {
     public function addConfig(array $config)
     {
-        $policyMap = [];
         $previousPolicyMap = [];
 
+        if ($this->containerBuilder->hasParameter('ezpublish.api.role.policy_map')) {
+            $previousPolicyMap = $this->containerBuilder->getParameter('ezpublish.api.role.policy_map');
+        }
+
         // We receive limitations as values, but we want them as keys to be used by isset().
         foreach ($config as $module => $functionArray) {
             foreach ($functionArray as $function => $limitationCollection) {
-                $policyMap[$module][$function] = array_fill_keys((array)$limitationCollection, true);
-            }
-        }
+                if ($this->policyExist($previousPolicyMap, $module, $function) && null !== $limitationCollection) {
```

Comment:
I would reverse this condition to `null !== $limitationCollection && $this->policyExist(...)`